sys/include/net/netbuf.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netbuf Network buffer abstraction
+ * @ingroup     net
+ * @brief       Provides an abstraction for network stack buffers
+ * @{
+ *
+ * @file
+ * @brief       Network buffer abstraction definition
+ *
+ * @author  Jose I. Alamos <jose.alamos@haw-hamburg.de>
+ */
+#ifndef NET_NETBUF_H
+#define NET_NETBUF_H
+
+#include <stdbool.h>
+#include "iolist.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Netbuf descriptor
+ */
+typedef void netbuf_ctx_t;
+
+/**
+ * @brief   Allocate a network buffer with a given size.
+ *
+ * @note    Supposed to be implemented by the networking module.
+ *
+ * @param[in]       size    size of the buffer to be allocated.
+ *
+ * @return          pointer to the netbuf with the allocated buffer.
+ * @return          NULL if there's not enough memory for allocation.
+ */
+netbuf_ctx_t *netbuf_alloc(size_t size, void **buf);
```

Comment:
This signature does not match the documentation and it seems weird to me, that the actual buffer part is an out parameter, while (potentially optional) context information is the return value. This should be the other way around, IMHO.

Reply:
Oh sorry, I forgot to update the documentation.
I can change it to what you propose. However, the ctx MUST be set (otherwise the network stack has no idea if the packet was allocated by itself or if it resides in a framebuffer)

Reply:
Addressed in https://github.com/RIOT-OS/RIOT/pull/12741/commits/2eb4b52ddcd9b30fea4d4cf277313259001c3830